### ISSUE TITLE
v7.21.0 is causing intellisense issues in VS code

### ISSUE THREAD

**Issue by fwaris** [ionide/ionide-vscode-fsharp#2035]
Just upgraded to v7.21.0. 

Seeing red squiggles in files that should compile fine.
Hovering over a squiggle gives 'index out of bounds' error message.

Fixed by going back to v7.20.3

Running Window 11 with the following sdks installed

```
C:\Users\Faisa>dotnet --list-sdks
6.0.416 [C:\Program Files\dotnet\sdk]
8.0.301 [C:\Program Files\dotnet\sdk]
8.0.400 [C:\Program Files\dotnet\sdk]
```

**Comment by vzarytovskii** [comment]
A stacktrace would be helpful, to understand if it happens in FSAC or FCS.

**Comment by vzarytovskii** [comment]
This is how it looks:
![telegram-cloud-photo-size-2-5343543798967429993-y](https://github.com/user-attachments/assets/070ad33e-b6b9-471e-8c50-cbad796b5c8f)

And also this:
![telegram-cloud-photo-size-2-5343543798967429995-y](https://github.com/user-attachments/assets/f26fd76d-0400-40b4-bcf5-7a340324611e)

**Comment by kant2002** [comment]
For the record. There no logs anywhere to be found. Even after enabling 
```
"FSharp.verboseLogging": true,
"FSharp.trace.server": "verbose",
```

It appear to me when developing using Oxpecker library, and when I modify computation expression. It's hard to reproduce for now. so does computatoin expression is the culprit is speculative.

**Comment by vzarytovskii** [comment]
Likely suspect is https://github.com/dotnet/fsharp/pull/17096/files which uses HashMultiMap as an underlying structure, which is not threadsafe.

**Comment by fwaris** [comment]
With the latest version -v7.21.1 - there are some issues still remaining - prob. related to the same fix.

When I edit a file, sometimes I get red squiggles. 
The error is: 
```
internal error: Operations that change non-concurrent collections must have exclusive access. A concurrent update was performed on this collection and corrupted its state. The collection's state is no longer correct. F# Compiler[193](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/compiler-messages/fs0193)
```

**Comment by vzarytovskii** [comment]
> With the latest version -v7.21.1 - there are some issues still remaining - prob. related to the same fix.
> 
> 
> 
> When I edit a file, sometimes I get red squiggles. 
> 
> The error is (when hovering over the squiggles): 
> 
> ```
> 
> internal error: Operations that change non-concurrent collections must have exclusive access. A concurrent update was performed on this collection and corrupted its state. The collection's state is no longer correct. F# Compiler[193](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/compiler-messages/fs0193)
> 
> ```
> 
> 

This should've been fixed in FCS already

**Comment by baronfel** [comment]
We're using the 8.0.400 build - did the fix come out after?

**Comment by vzarytovskii** [comment]
> We're using the 8.0.400 build - did the fix come out after?

Yes, and not in 8 branch, we didn't backport it there

**Comment by baronfel** [comment]
Ok, so FSAC updating to https://www.nuget.org/packages/FSharp.Compiler.Service/43.9.100-preview.24422.2 or similar should unblock this.

**Comment by baronfel** [comment]
I suppose now that RC1 is out we should probably look at doing this anyway.

**Comment by vzarytovskii** [comment]
Yeah, there are whole bunch of fixes there. Should be fine despite it being a "preview".
